src/_includes/site_navigation.html

```diff
@@ -0,0 +1,34 @@
+{% assign sections = site.pages | site_sections %}
+{% assign parent_page = site.pages | parent_of: page %}
+
+<aside class="related">
+  <nav aria-label="Site navigation">
+    <ul>
+      {% for section in sections %}
+          <li class="related__section-link">
+            <a href="{{ section.url | relative_url }}" {% if section.url == page.url %}class="current" aria-current="page"{% endif %}>{{ section.title }}</a>
+              {% assign grouped_related_pages = site.pages | children_of: section | group_by: "related_order" | sort: "name" %}
+              {% if grouped_related_pages.size > 0 %}
+                <label class="visually-hidden" for="section-{{ forloop.index }}">Toggle page links in this section</label>
```

Comment:
Rather than "in this section" this should be something like "in {{ section.title }}" so that the list of controls read out in a screen reader rotor/elements list makes sense out of context and is fully WCAG compliant